examples/oauth/src/main.rs

```diff
@@ -256,3 +269,30 @@ where
         Ok(user)
     }
 }
+
+// Use anyhow, define error and enable '?'
+// For a simplified example of using anyhow in axum check /examples/anyhow-error-response
+#[derive(Debug)]
+struct AppError(anyhow::Error);
+
+// Tell axum how to convert `AppError` into a response.
+impl IntoResponse for AppError {
+    fn into_response(self) -> Response {
+        (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            format!("Something went wrong: {}", self.0),
```

Comment:
I don't think we should return the error here. That could reveal private details about the application. Instead we should return a generic "Something went wrong" body and log the error.

I also think the error should be logged as `tracing::error!("{:#}", self.0)`, specifically using `{:#}` as documented [here](https://docs.rs/anyhow/latest/anyhow/struct.Error.html#display-representations).

Reply:
Solid feedback, appreciated!